### PR TITLE
Add TPSV, refactor TRSV and TBSV Level 2 BLAS Operators

### PR DESCRIPTION
**Pull request by pgorlani**
This patch adds an implementation of `TPSV` Level 2 operator compatible with `TUNING_TARGET` specializations for `INTEL_GPU`, `AMD_GPU` and `NVIDIA_GPU`.

Moreover, this patch refactors and unifies the implementations of `TRSV` and `TBSV` with `TPSV`.

### CODE REVIEW

**Review by aacostadiaz — COMMENTED** [review]
Nice refactor! I just left a comment to initialise the labels in the benchmark.

**Review by aacostadiaz — APPROVED** [review]

**Review by aacostadiaz — APPROVED** [review]
Approval for the recent changes.

File: src/interface/blas2_interface.hpp [hunk 1]
```diff
@@ -350,15 +352,16 @@ typename sb_handle_t::event_t _trsv_impl(sb_handle_t& sb_handle, index_t _N,
       blas::make_sycl_iterator_buffer<int32_t>(sync_vec, sync_vec.size());
   auto sync = make_vector_view(sync_buffer, one::value(), sync_vec.size());
 
-  auto trsv =
-      make_trsv<subgroup_size, subgroups, is_upper, is_transposed, is_unit>(
-          vx, mA, vx, sync);
+  index_t unused;
```
Comment (mehdi-goli) [inline src/interface/blas2_interface.hpp:355]: Why do we introduce the unused variable. What was wrong with the previous definition. It is better to use the previous definition here as it was more readable and obvious that adding unused variable and making them to use the same function

Comment (pgorlani) [reply src/interface/blas2_interface.hpp]: Done in https://github.com/codeplaysoftware/sycl-blas/pull/434/commits/441067da63570046d6ce44e9f55a971ee9246927.

**Review by mehdi-goli — APPROVED** [review]

**Pull request merged**